### ISSUE TITLE
Move textarea for code to an actually webeditor

### ISSUE THREAD

**Issue by NuttyShrimp** [ZeusWPI/ledstrip_sandbox#10]
The code for ledstrip is currently shown/written in a basic textarea. I think it would be a great idea to replace this textarea with a small embedded editor (eg. aceditor)

**Comment by rien** [comment]
There has been quite a bit of research around browser editors at Dodona for the [Papyros](https://papyros.dodona.be/) and [Dolos](https://dolos.ugent.be) projects. The conclusions were roughly the following:
- Ace editor (which Dodona currently uses) is minimalist and not ideal
- [Monaco](https://microsoft.github.io/monaco-editor/) (the editor powering vscode) has a lot of features, but lacks modularity, this is currently used in [Dolos](https://dolos.ugent.be)
-  [CodeMirror](https://codemirror.net/) (which Papyros currently uses) is very modular and has good documentation

To summarize: if you want a full-fledged editor with little effort, use Monaco, if you want to tinker with it (e.g. implement debugging and whatever) use CodeMirror.

@chvp may also have good opinions & ideas about this

**Comment by NuttyShrimp** [comment]
Personally I would vote for monaco because it is really easy to implement and has an [unpkg source](https://unpkg.com/browse/monaco-editor@0.34.0/) which redfast wanted
